apps/src/aichat/views/modelCustomization/PublishNotes.tsx

```diff
@@ -24,24 +29,34 @@ const PublishNotes: React.FunctionComponent = () => {
     state => state.aichat.currentAiCustomizations
   );
 
-  const onUpdate = useCallback(
-    () => dispatch(updateAiCustomization()),
-    [dispatch]
-  );
+  const onSave = useCallback(() => {
```

Comment:
These multiple dispatches I believe are not the right way to do this, so looking for input :)

I see we have a couple instances where we put some redux updates that we want to happen after some async operation in this structure:

https://github.com/code-dot-org/code-dot-org/blob/194f7f04771c9cc1fab64fbcc598585ee8406bcd/apps/src/aichat/redux/aichatRedux.ts#L318-L331

We also dispatch some redux actions inside of the thunk here:

https://github.com/code-dot-org/code-dot-org/blob/194f7f04771c9cc1fab64fbcc598585ee8406bcd/apps/src/aichat/redux/aichatRedux.ts#L120-L139

Is one of these preferable? Is there a difference? Skimming through docs here, it seems like the first approach might be more standard?

https://redux-toolkit.js.org/api/createAsyncThunk

Reply:
Seems like either approach is reasonable but defer to @sanchitmalhotra126 on a preference.

Reply:
Yeah, I mean nothing wrong with back to back dispatches as such, but if the goal here is that we only update the publish state after saving has completed, we should move this into the async thunk. In fact, if we need to save the publish state to the project, it might actually be better to do this in the opposite order - on clicking save, if the model card is not filled out, set hasPublished to false so that when you do save, that value is saved to the project.

Reply:
Check out the updates in the last two commits...there's at least one thing to figure out (I have an `any` in there 😬 ), but I tried to refactor to use async thunks. I don't see the approach I have here (a shared bit of async code that takes in `thunkAPI` as an arg), which usually isn't good. LMK what you think!